hnn_core/tests/test_mpi_simulation.py

```diff
@@ -0,0 +1,61 @@
+import os.path as op
+
+from numpy import loadtxt
+from numpy.testing import assert_array_equal
+
+from mne.utils import _fetch_file
+import hnn_core
+from hnn_core import Params
+
+
+def test_simulate_dipole_mpi():
+    """Test to check if simulate dipole can be called with MPI."""
+    from mpi4py import MPI
+    from psutil import cpu_count
+
+    # need to get physical CPU cores using psutil and leave one CPU
+    # for the current script (bug in OpenMPI < 4.0) or MPI will fail
+    # to spawn nrniv workers
+    n_core = cpu_count(logical=False) - 1
+    if n_core < 1:
+        n_core = 1
+
+    # spawn NEURON sim
+    args = ['-python', '-mpi', '-nobanner', 'python',
+            op.join(op.dirname(hnn_core.__file__), 'tests',
+                    'test_compare_hnn.py')]
```

Comment:
is this the only way to use MPI? can it not be called from within python without having to invoke a command line argument?

Reply:
This is the best way to test MPI. If you start an MPI native script, you have to case everything with `if rank == 0` and the like. This way, the same script for testing hnn can be used. Note that the same thing (invoking another process) inevitably happens with joblibs (spawn multiple processes on the operating system). It's just a little more transparent with MPI because it looks like a Popen or similar.

Reply:
For the non-pythonic way, but how I would start it on a cluster see below:
https://github.com/hnnsolver/hnn-core/pull/79/files#diff-cbfdde8e3e8b0f3ad7c8ab1ffce9a9c7R33-R45

Reply:
@jasmainak see my recent comment below that nrniv must be invoked

Reply:
This complexity is hidden with `MPI_backend`!!